specs/agents/configuration.md

```diff
@@ -43,15 +43,16 @@ agents:
 | Boolean  | Encoded as a lower-case boolean string: `"false"`, `"true"` |
 | List     | Encoded as a comma-separated string (whitespace surrounding items should be stripped): `"foo,bar,baz"` |
 | Mapping  | Encoded as a string, with `"key=value"` pairs separated by commas (whitespace surrounding items should be stripped): `"foo=bar,baz=foo"` |
-| Duration | String with duration encoded using unit suffixes (`ms` for millisecond, `s` for second, `m` for minute). |
+| Duration | String with duration encoded using unit suffixes (`ms` for millisecond, `s` for second, `m` for minute). Validating regex: `^(-)?(\d+)(ms|s|m)$` |
+| Size     | String with a positive size encoded using unit suffixes (`b` for bytes, `kb` for kilobytes, `mb` for megabytes, `gb` for gigabytes). Validating regex: `^(\d+)(b|kb|mb|gb)$` |
```

Comment:
Should we specify Size as being case-insensitive? Currently all the agents that have size config options (all but .NET and PHP) are case-insensitive here.

(I'm happy to be told I'm being over-pedantic here! My last gig was with a cloud storage solution where the difference between a petabyte, PB, and a pebibyte, PiB, was significant.) Should we specify these as kibibytes, mebibytes, and gibibytes (https://en.wikipedia.org/wiki/Byte#Multiple-byte_units) which denote  using 1024 (base 2) as the multiplier? Currently all agents, **except Ruby notably**, use 1024 as the multiplier.

The ruby exception: https://github.com/elastic/apm-agent-ruby/blob/d8230327187f38ce24f51f679ef7a192aab4efd6/lib/elastic_apm/config/bytes.rb#L24-L33
- It allows eliding the unit and defaults to "kb".
- It uses an odd mix of 1024 and 1000 multipliers.
- I think the `100_000` multiplier for "gb" is a bug (should be `1_000_000` or `1024 * 1024`)

Reply:
Yes, both size and duration should be case-insensitive, and we should put that in this table.

On prefixes for sizes: We should define the multiplier. As far as kilo vs kibi, I'm not sure it's worth the change, especially because at your wikipedia link it specifically shows that kilo/mega/giga have been used for 1024 multipliers when talking about memory. It does list them as "deprecated" but I don't think there's going to be any confusion as long as we define the multiplier.

Anyone else have any input here?

Reply:
+1 on defining the multiplier being 1024
+1 on keeping mb vs mib
+1 on making durations case-sensitive
+1 on making size suffixes case-insensitive